### PR TITLE
Longest substring ch20

### PR DESCRIPTION
**Pull request by Yuvraj-kadale**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?
YES

### Description

Added longest substring without repeating characters in C, having a linear time complexity [ O(n) ] and constant space Complexity O(1). 

||Output||
Capture

--Some other input-output---
Capture

Capture

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

issue #563 Solved !!

### CODE REVIEW

**Review by iamrajiv — CHANGES_REQUESTED** [review]
@Yuvraj-kadale add sample input-output, time complexity and space complexity

**Review by iamrajiv — CHANGES_REQUESTED** [review]
@Yuvraj-kadale add sample input-output, time complexity and space complexity in `Longest substring without repeating chars.cpp` program in the form of comments. Also, move this program to `C/cp` not in `C/other`.

**Review by Abhijit2505 — CHANGES_REQUESTED** [review]
Your code is good, but you need to place your code in the proper place as reviewed by @iamrajiv .

**Comment by Abhijit2505** [comment]
Please work upon the merge conflict thing, make a pull from the master repo to the forked repo.

**Review by iamrajiv — CHANGES_REQUESTED** [review]
@Yuvraj-kadale Do not edit `.all-contributorsrc` and `CONTRIBUTORS.md ` files. These files are managed by All Contributors bot.

File: C/other/Longest substring without repeating chars.cpp [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+//Input: Hackincodes
+//Output: length of longest substring without repeating charecters: 7
+
+//Input: ddddd
+//Output: length of longest substring without repeating charecters: 1
+
+//Input: 
+//Output: length of longest substring without repeating charecters: 0
+------------------------------------------------------------------------------
```
Comment (iamrajiv) [inline C/other/Longest substring without repeating chars.cpp]: @Yuvraj-kadale Remove these type of line patterns.

File: C/other/Longest substring without repeating chars.cpp [hunk 2]
```diff
@@ -0,0 +1,77 @@
+//This code is submitted to NeoAlgo as a part of Contributors hack 2020
```
Comment (iamrajiv) [inline C/other/Longest substring without repeating chars.cpp]: @Yuvraj-kadale Make all single-line comments in one multiline comment. Something like this
```
/*
...
*/
```

**Review by iamrajiv — CHANGES_REQUESTED** [review]

File: C/other/Longest substring without repeating chars.cpp (hunk 1 above)
Comment (Yuvraj-kadale) [reply C/other/Longest substring without repeating chars.cpp]: Done removed them!!

File: C/other/Longest substring without repeating chars.cpp (hunk 2 above)
Comment (Yuvraj-kadale) [reply C/other/Longest substring without repeating chars.cpp]: Done!!

**Comment by Yuvraj-kadale** [comment]
> 
> 
> @Yuvraj-kadale add sample input-output, time complexity and space complexity in `Longest substring without repeating chars.cpp` program in the form of comments. Also, move this program to `C/cp` not in `C/other`.

Done, file has been moved to C/cp from C/others

**Comment by Yuvraj-kadale** [comment]
> 
> 
> Please work upon the merge conflict thing, make a pull from the master repo to the forked repo.

I just saw in the PR the merge conflicts are due to updation in the readme files and contributors files!
https://github.com/TesseractCoding/NeoAlgo/pull/902#pullrequestreview-486961867

should I send a new PR!? and close this PR!?

**Comment by atarax665** [comment]
@Yuvraj-kadale I am closing this PR as it has so many conflicts, open a new PR for this issue.

**Pull request closed without merging**